custom_components/hacs/helpers/classes/repositorydata.py

```diff
@@ -66,7 +67,21 @@ def name(self):
 
     def to_json(self):
         """Export to json."""
-        return attr.asdict(self)
+        return {
```

Comment:
we can just reuse the custom JSON encoder that core uses here instead right?

Reply:
This one is a tricky as this is what would would be the 'as_dict' step in core before it's serialized to json.   

Maybe I should rename the function as it a bit confusing and doesn't show the exact intent.

Reply:
Another way would be to round-trip to json, and then compare.  Its slightly more cpu overhead, but these files are small and it would be more forward compatible.

Reply:
That is the approach I had in mind, I think the added overhead is worth it here

Reply:
I agree, will be a lot easier to maintain going forward.  I'll add a `async_save_to_store_pure` that doesn't pass it back through `JsonEncoder` if we round trip it so we don't have to pay the overhead twice.